library/js/xl/translate.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Javascript translate function.
+ *  This calls the i18next.t function that has been set up in main.php
+ *
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ * @author  Brady Miller <brady.g.miller@gmail.com>
+ * @copyright Copyright (c) 2019 Brady Miller <brady.g.miller@gmail.com>
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+function xl(string) {
+    if (typeof top.i18next.t == 'function') {
+        // top
+        return top.i18next.t(string);
+    } else {
+        // opener (if called from a modal/popup)
+        return opener.i18next.t(string);
+    }
```

Comment:
@sjpadgett , I'm a bit lost on what I did above. It looks like the 2 popup/modal testing points above work even if I change the statement in the else to `return top.i18next.t(string);`. Am I overcomplicating this?

Reply:
disregard. sorted it out. above is working everywhere so far.

Reply:
Instead of having a separate script source file that will probably never have new translate functions added, could we not start a javascript api include script where we could add various js utility functions?
Start it like a library include file.

Reply:
sure, I'll change it to utility.js unless you have a better name